src/browser/deck/actions.js

```diff
@@ -90,83 +96,136 @@ define((require, exports, module) => {
     switchSelected(items, selected(items), item);
 
-  // Takes `items` and selects item previous to currently selected one.
-  // If selected item is first item, then last item is selected.
+  //  Makes an item leading a `selected` item `selected`.
   const selectPrevious = advance(switchSelected,
                                  selected,
                                  previous);
+
+  // Make an item following a `selected` item `selected`.
   const selectNext = advance(switchSelected,
                              selected,
                              next);
 
-  // Takes deck `items` and activates selected item.
+  // Makes `active` item `selected`.
+  const reset = items => select(items, active(items));
+
+  // Makes `selected` item `active`.
   const activate = advance(switchActive,
                            active,
                            selected);
 
-  // Resets currently selecetd tab back to the active one.
-  const reset = items => select(items, active(items));
-
-
-  const MAX_ORDER = Number.MAX_SAFE_INTEGER;
-  const orderOf = item => item.get('order') || MAX_ORDER / 2;
-  const order = items => items.sortBy(orderOf);
+  // Makes an item following a `selected` item both `selected` & `active`.
+  const activateNext = compose(activate, selectNext);
+  // Makes an item leading a `selected` item both `selected` & `active`.
+  const activatePrevious = compose(activate, selectPrevious);
 
-  // Returns `item` that when ordered will be in `n`th position
-  // starting from `0`.
-  const nth = (items, n) => items.sortBy(orderOf).get(n);
 
-  const asNth = (items, n, item) => {
-    const ordered = order(items);
-    const before = items.get(n);
-    const after = items.get(n + 1);
-    const order = ((after - before) / 2) + before;
-    return item.set('order', order);
+  // ## Item weight
+  //
+  // Order of `items` in the list in most cases is irrelevant, in fact any
+  // item inserted is added to the end of the list. In order to allow ordering
+  // though notion of `weight` is used. Each item is assigned a weight during
+  // insertion that is a number right in the middle of leading and following
+  // item weights. In order to move item into arbitrary position it's weight
+  // is updated again to the number that is weight in between leading and
+  // following items. Items get reordered post-activation to position recently
+  // activated items to the head.
+
+  const MAX_WEIGHT = Number.MAX_SAFE_INTEGER;
+  const MIN_WEIGHT = 0;
+
+  // Returns weight for the given item.
+  const weightOf = item => item.get('weight') || (MIN_WEIGHT + MAX_WEIGHT) / 2;
+
+  // Returns items sorted by their weight.
+  const arrange = items => items.sortBy(weightOf);
+
+  const center = (from, to) => (from + to) / 2;
+
+  // Returns given `item` with modified `weight` such that it's
+  // weight will fall in the center of `leading` & `following`
+  // item weights.
+  const asBetweenOf = (item, leading, following) => {
```

Comment:
Idea: `weightBetween(item, leading, following)` or `setWeightBetween(item, leading, following)` would be clearer naming scheme.